Sources/Datadog/RUM/Debugging/RUMDebugging.swift

```diff
@@ -0,0 +1,185 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+internal protocol RUMDebugging {
+    func debug(applicationScope: RUMApplicationScope)
+}
+
+#if targetEnvironment(simulator)
```

Comment:
❔  you probably considered `#if DEBUG` too, why not `#DEBUG`?

Reply:
Some apps may provide different setup in `RELEASE` configuration or even use custom `DEBUG` configuration name, so my consideration was: `#if targetEnvironment(simulator)` is more flexible.

Reply:
in order for us to be on the same page: 
we should allow debugging in non-production builds and disallow in production builds, no?

btw you can use `#if DEBUG` with different config names (as long as you don't remove default `DEBUG` active compilation condition)

Reply:
Actually, after talking with @jegnux, we concluded that for our mobile app it is valid to use `debugRUM = true` even in non `DEBUG` builds, so it can be toggled in Test Flight or even App Store builds with Debug Assistant view. As this may be also the use case for other apps, WDYT about removing `#if` at all, @buranmert? Then, this would be up to the user to ensure it's not enabled in production builds when not necessary. Less safety / more control.

Reply:
yes, then non-devs can test the app with this 👍 

but i'm still hesitating to add it to public api? we can still provide it in a semi-public way with `NotificationCenter.post` for example?
do we want to support this feature in tvOS, SwiftUI, React Native, etc.?

Reply:
TBH I don't understand the idea with `NotificationCenter.post`, how would it work? To me those are clear enough:
```swift
/// One debug utility:
Datadog.verbosityLevel

/// Another debug utility:
Datadog.debugRUM
```

Reply:
```swift
NotificationCenter.default.post(name: .init("datadog.enableRUMDebugging"), object: true/false)
```
this is a raw idea, we can mention this notification name in docs or readme (assuming if they are debugging the SDK then they probably look at docs and/or readme first)
the idea is if we decide to drop it from public API, we can do it silently without breaking their builds
wdyt?

btw i found out that `verbosityLevel` was a debug utility thanks to a GH issue, it may not be clear to everyone

Reply:
I don't see how the notification may be discoverable and TBH I never saw the API like that. Where we would put it in the documentation?

> btw i found out that verbosityLevel was a debug utility thanks to a GH issue, it may not be clear to everyone

The code documentation states it clear that it's for debugging purposes.
```swift
/// Verbosity level of Datadog SDK. Can be used for debugging purposes.
/// If set, internal events occuring inside SDK will be printed to debugger console if their level is equal or greater than `verbosityLevel`.
/// Default is `nil`.
public static var verbosityLevel: LogLevel? = nil
```

Reply:
As concluded on Slack - let's go with `Datadog.debugRUM` and we may consider adding `ProcessInfo`-based configuration later.